src/api/appcenter_api.rs

```diff
@@ -71,4 +71,32 @@ impl API for AppCenter {
             Err(_e) => Err("Failed to fetch error groups json"),
         }
     }
+
+    fn os_versions(
+        &self,
+        organization: &str,
+        application: &str,
+        error_group_id: &str,
+    ) -> Result<String, &'static str> {
+        let url = format!(
+            "https://api.appcenter.ms/{}/apps/{}/{}/errors/errorGroups/{}/operatingSystems",
+            API_VERSION, organization, application, error_group_id
+        );
+        let response = self
+            .client
+            .get(&url)
+            .header("X-API-Token", self.token.clone())
+            .header("accept", "application/json")
+            .send();
+        match response {
+            Ok(mut response) => {
+                let mut json = String::new();
+                match response.read_to_string(&mut json) {
+                    Ok(_j) => Ok(json),
+                    Err(_e) => Err("Failed to read response from API"),
+                }
+            }
+            Err(_e) => Err("Failed to fetch error groups json"),
```

Comment:
It's not as straightforwards, but I'll take a look. `Err` can't format the string, as written, by itself.

Reply:
So this doesn't work? 
```
use std::fs::File;

fn main() {
    let f = File::open("hello.txt");

    let f = match f {
        Ok(file) => file,
        Err(error) => panic!("Problem opening the file: {:?}", error),
    };
}
```
I assume the Elvis operator is there to handle optional chaning? https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html

Reply:
I think we have this somewhere in the code base too

Reply:
The Elvis operator is there for extended debug output of the error object.
At least the compiler also says that `{}` is fine.

I've added the `panic!` macro, as suggested. That should be fine, since without responses from AppCenter we may not want to continue execution.